unittests/savanna_misc_tests.cpp

```diff
@@ -688,5 +687,106 @@ BOOST_FIXTURE_TEST_CASE(verify_spring_1_0_block_compatibitity, savanna_cluster::
 
 } FC_LOG_AND_RETHROW()
 
+/* -----------------------------------------------------------------------------------------------------
+            Finality advancing past block claimed on alternate branch
+            =========================================================
+
+Time:        t1      t2      t3      t4      t5      t6      t7      t8
+Blocks:
+    B0 <---  B1 <--- B2 <--- B3 <-|- B4 <--- B5
+                                  |
+                                  \----------------- B6 <--- B7 <--- B8 <--- B9
+QC claim:
+           Strong          Strong  Strong  Strong  Strong   Weak   Strong  Strong
+             B0              B1      B3      B4      B2      B6      B7      B8
+
+Vote:                      Strong  Strong   Strong  Weak   Strong  Strong  Strong
+
+                                                     ^
+                                                     |
+                                                 validating those weak votes on b2
+                                                 would fail if nodes have received b4 and b5
+                                                 which advanced lib to b3
+
+    - Node D is isolated and has not seen B3, B4, and B5
+    - it received B3 via push_block, (so it can make it its head and produce a child of B3), but has not
+      received votes on b3 (only on b2), so b6 includes a strong QC on b2.
+    - when b6 is pushed to A, B and C, qc validation would fail in Spring 1.0.0 because the fork_db
+      lookup, but succeeds in Spring 1.0.1 as only the finality core of the claiming block is needed
+      to validate the QC.
```

Comment:
This comment needs to be updated for this test.

Reply:
You're right, thanks. I also updated the PR description.

Reply:
https://github.com/AntelopeIO/spring/pull/782/commits/78c1053e75ccde144fc424429c38e4800d760aa9